test/contrib/test_socks.py

```diff
@@ -199,6 +206,8 @@ class TestSocks5Proxy(IPV4SocketDummyServerTestCase):
     Test the SOCKS proxy in SOCKS5 mode.
     """
     def test_basic_request(self):
+        disable_warnings(NoIPv6Warning)
```

Comment:
py.test restores the default warnings filters after every run. Here’s a minimal example:

```python
import warnings

from urllib3 import disable_warnings
from urllib3.exceptions import InsecureRequestWarning


def test1():
    print('test1 before', len(warnings.filters))
    disable_warnings(InsecureRequestWarning)
    print('test1 after', len(warnings.filters))


def test2():
    print('test2 before', len(warnings.filters))
    disable_warnings(InsecureRequestWarning)
    print('test2 after', len(warnings.filters))
```

which has output:

```
('test1 before', 9)
('test1 after', 10)
('test2 before', 9)
('test2 after', 10)
```

If you print the filters rather than their length, you can see the `InsecureRequestWarning` filter is purged between runs.

As to ignoring them entirely – I wonder if putting them in `conftest.py` would do it? I’ll try that.